test/cpp/api/transformer.cpp

```diff
@@ -460,3 +430,492 @@ TEST_F(TransformerTest, PrettyPrintTransformerDecoderLayer) {
       "  (norm3): torch::nn::LayerNorm([4], eps=1e-05, elementwise_affine=true)\n"
       ")");
 }
+
+void transformer_decoder_test_helper(bool is_cuda) {
+  // this is a deterministic test for TransformerDecoder
+  torch::Device device = is_cuda ? torch::kCUDA : torch::kCPU;
+  torch::TensorOptions tensor_options =
+    torch::TensorOptions().dtype(torch::kFloat32).device(device);
+
+  TransformerDecoderLayer decoder_layer = get_a_test_layer<
+    TransformerDecoderLayer,
+    TransformerDecoderLayerOptions>(tensor_options);
+
+  TransformerDecoder model(TransformerDecoderOptions(decoder_layer, 1));
+  if (is_cuda) {
+    model->to(torch::kCUDA);
+  }
+
+
+  at::Tensor decoder_input = torch::tensor({{{20, 30, 40, 50}}},
```

Comment:
use torch::Tensor instead of at::Tensor for the following lines. at scope is internal.